src/tracing/builder/parity.rs

```diff
@@ -413,13 +423,20 @@ where
 
     fn next(&mut self) -> Option<Self::Item> {
         // ensure the selfdestruct trace is emitted just at the ending of the same depth
-        if let Some(selfdestruct) = &self.next_selfdestruct {
-            if self
-                .iter
-                .peek()
-                .is_none_or(|(next_trace, _)| selfdestruct.trace_address < next_trace.trace_address)
-            {
-                return self.next_selfdestruct.take();
+        if !self.next_selfdestructs.is_empty() {
+            // find the next selfdestruct to yield
+            if let Some((next_trace, _)) = self.iter.peek() {
+                // find the most recently recorded selfdestruct that has a lower address
+                if let Some(pos) =
+                    self.next_selfdestructs.iter().rev().position(|selfdestruct| {
+                        selfdestruct.trace_address < next_trace.trace_address
+                    })
+                {
+                    return Some(self.next_selfdestructs.remove(pos));
```

Comment:
`pos` is not an index into the vec but a count from the end

Reply:
right